httpclient-vertx/src/main/java/io/fabric8/kubernetes/client/vertx/VertxHttpClientFactory.java

```diff
@@ -76,6 +76,9 @@ public VertxHttpClient build() {
 
       WebClientOptions options = new WebClientOptions();
 
+      options.setMaxPoolSize(1024);
```

Comment:
Are there any downsides to leaving this unlimited?  We didn't assume any limit for the other implementations.

Reply:
The downside is a potential resource exhaustion on the client, but also on the origin server. It is up to you to decide how you want to do that.

Note that the Jetty server has a max value but does not enforce it by default.

Reply:
> The downside is a potential resource exhaustion on the client, but also on the origin server. It is up to you to decide how you want to do that.

We'll just assume unlimited, then we can later limits should anyone experience a problem - but we'll probably want to put that logic in the StandardHttpClient such that it's common to all http clients and make it as proactive as possible to inform  the user exactly what is happening.

> Note that the Jetty server has a max value but does not enforce it by default.

These tests are using com.sun.net.httpserver.HttpServer and the okhttp3.mockwebserver.MockWebServer at least up to 1024 connections they don't seem to require any additional configuration either.

Reply:
the issue might arise when multiple clients are targeting the same server with the same level of concurrency, but I don't know about your use cases.

Reply:
I'm not sure is the responsibility of the (Kubernetes) client to protect a Kubernetes server->cluster from being flooded.

I understand this might be concerning for any standard use-case where the target server is a regular HTTP server. However, in this case, the API server should be strong enough to handle or scale when a given number of concurrent connections are performed (Come on, it's Kubernetes after all ;))

The decision to remove the hard limit on the client-side was to achieve a consistent behavior across all `HttpClient` implementation with the minimum effort.